src/main/scala/loamstream/googlecloud/CloudSdkDataProcWrapper.scala

```diff
@@ -0,0 +1,92 @@
+package loamstream.googlecloud
+
+import loamstream.util.Loggable
+import loamstream.util.Paths.normalize
+import scala.util.Try
+import scala.util.Success
+import loamstream.util.Tries
+import loamstream.util.ExitCodes
+import scala.concurrent.duration.Duration
+import loamstream.util.BashScript
+import java.nio.file.Paths
+
+/**
+ * @author clint
+ * Nov 28, 2016
+ */
+final class CloudSdkDataProcWrapper private[googlecloud] (
+  config: GoogleCloudConfig,
+  runCommand: Seq[String] => Int = CloudSdkDataProcWrapper.runCommand) extends 
+      DataProcClient.CanStop with DataProcClient.CanTellIfRunning with Loggable {
+
+  import CloudSdkDataProcWrapper.{
+    deleteClusterTokens,
+    isClusterRunningTokens
+  }
+
+  override def stopCluster(): Unit = {
+    debug(s"Deleting cluster '${config.clusterId}'")
+
+    val exitCode = runCommand(deleteClusterTokens(config))
+
+    ExitCodes.throwIfFailure(exitCode)
+  }
+
+  override def isClusterRunning: Boolean = {
+    val exitCode = runCommand(isClusterRunningTokens(config))
+
+    val result = ExitCodes.isSuccess(exitCode)
```

Comment:
So, this bit of code took me a bit to grok. It's asking for `isClusterRunning`, but then checking `exitCode` for success, which instantly made me think it's done running. After re-examining and looking at the surrounding functions, it becomes more clear what's going on: run an external command to check for status and then testing the exit code of the external command.

I just wonder if this could be done a little differently to make it a bit more clear at first glance? Possible ideas (not trying to indicate they're better, just throwing stuff at the wall...):

Make the `runCommand` more prominent as a `match`?

```scala
runCommand(isClusterRunningTokens(config)) match {
  case Success(_) => true
  case _ => false
}
```

Allow the return type of `runCommand` to specify whether or not it was successful?

```scala
runCommand(isClusterRunningTokens(config)).isSuccess
```

If the latter is appealing, I wonder if it would be more clear to allow the tokens to run themselves then?

```scala
isClusterRunningTokens(config).run().isSuccess
```

Anyway, just tossing out ideas.

Reply:
I'll see what I can do here.  `ExitCodes` is an old class, that attempts to abstract over checking exit codes, since at one point we were dealing with oddball tools that didn't follow the "0 means success" convention.  That doesn't apply anymore, and I've been trying to move away from `ExitCodes` and invoking `scala.sys.process` stuff directly in favor of `loamstream.util.Processes.runSync` - which would be close to what you're proposing here, but there are a few subtly-different use cases to reconcile.

Reply:
I'm going to defer this to another PR.  I tried a few variations here and couldn't come up with something that didn't introduce a ton of code, or provided a good bang for the buck.  Later, I'd still like to get rid of the direct invocation of `scala.sys.process` by `CloudSdkDataProcWrapper` in favor of one way to run processes provided by `loamstream.util.Processes`.